### PR TITLE
Add migration to regenerate derivatives on filesets with too many

### LINKED ISSUE #5352: fileset with too many derivatives
Is this right? https://figgy.princeton.edu/catalog/99bd0a3f-488c-4c36-8519-d75fb0a87b44

found while investigating errors from 8/4 on https://app.honeybadger.io/projects/53391/faults/84134442/01G9MZNGJDN3SD9FEJP9Z0FZ72#notice-summary

There were 587 errors on 8/4; one was a scanned resource image that needed a regenerated derivative.

Here's a list of ids: https://gist.github.com/hackartisan/9ee3224cc8b45cd9c4e1bb6e00b32317

### PR DESCRIPTION
**Pull request by hackartisan**
closes #5352

### CODE REVIEW

File: app/services/migrations/remove_extra_derivatives.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    private
+
+      def progress_bar
+        @progress_bar ||= ProgressBar.create format: "%a %e %P% Resources Processed: %c of %C", total: query_service.custom_queries.count_all_of_model(model: FileSet)
+      end
+
+      def extra_derivatives?(file_set)
+        file_set.file_metadata.count(&:derivative?) > 1
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/services/migrations/remove_extra_derivatives.rb:26]: The above doesn't really work. Query to get file set IDs with more than two derivatives:

```sql
select a.id, COUNT(file_metadata_element) FROM orm_resources a, jsonb_array_elements(a.metadata->'file_metadata') AS file_metadata_element WHERE file_metadata_element @> '{"use": [{"@id": "http://pcdm.org/use#ServiceFile"}]}' GROUP BY a.id HAVING(count(file_metadata_element)) > 2 LIMIT 10;
```

It's pretty slow, getting 10 takes a while.

File: app/services/migrations/remove_extra_derivatives.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+module Migrations
+  class RemoveExtraDerivatives
+    def self.call
+      new.run
+    end
+
+    def run
+      progress_bar
+      query_service.find_all_of_model(model: FileSet).each do |file_set|
```
Comment (tpendragon) [inline app/services/migrations/remove_extra_derivatives.rb:11]: This isn't memory efficient - it'll load every file set into memory. See https://github.com/samvera-labs/valkyrie-sequel/blob/main/lib/valkyrie/sequel/query_service.rb#L43-L47.

We created the MemoryEfficientAll query for reindexing, but that doesn't let us filter to a specific model - it just has a deny list.

File: app/services/migrations/remove_extra_derivatives.rb (hunk 1 above)
Comment (tpendragon) [inline app/services/migrations/remove_extra_derivatives.rb:26]: It's looking like > 2 is a decent check, I'm running a longer query to see if there's any that we wouldn't want to regenerate that have that count.

**Review by tpendragon — CHANGES_REQUESTED** [review]

File: app/services/migrations/remove_extra_derivatives.rb (hunk 1 above)
Comment (hackartisan) [reply app/services/migrations/remove_extra_derivatives.rb:26]: we could try limiting it to file sets on raster resources, the bug was in the raster resource derivative service

**Review by tpendragon — APPROVED** [review]

**Pull request merged**